beacon/blockchain/payload.go

```diff
@@ -236,3 +246,26 @@ func (s *Service) handleOptimisticPayloadBuild(
 
 	s.metrics.markOptimisticPayloadBuildSuccess(buildData.Slot)
 }
+
+func PrevBlockProposerPubKey[
+	ForkChainSpec interface {
+		ActiveForkVersionForTimestamp(timestamp math.U64) common.Version
+	},
```

Comment:
this generic seems necessary to reuse this function in both block building  package and here for optimistic block building

Reply:
nit: just for code cleanliness, can we define ForkChainSpec in interfaces.go? or better even, use the existing ServiceChainSpec already defined there

Reply:
The chain specs in blockchain and payload packages defines different methods and I currently cannot reuse/embed one into another. This is the minimum interface which both must implement to reuse the method.
I am not found of the generic either but don't currently see a better solution. Do you have suggestions.